src/containers/PatientDetails/PatientApps/__tests__/useSmartApps.test.ts

```diff
@@ -0,0 +1,134 @@
+    });
+    return ensureSave({
+        resourceType: 'Client',
+        type: clientType,
+        name: `${role} SmartForms`,
+        description: `${role} Smart Forms App`,
+        grant_types: ['authorization_code']
+    });
+}
+
+async function initialSetup(role: Role) {
+    const data = await dataSetup(role);
+    await login({ ...data.user, password: 'password' });
+    const client = await withRootAccess(axiosInstance, async () => {
+        return await createClient(role);
+    });
+    return { ...data, client };
+}
+
+function dataSetup(role: Role) {
+    return withRootAccess(axiosInstance, async () => {
+        let resource;
+        let resourceType;
+        switch (role) {
+            case Role.Patient:
+                resource = await createPatient();
+                resourceType = 'Patient';
+                break;
+            case Role.Practitioner:
+                resource = await createPractitioner();
+                resourceType = 'Practitioner';
+                break;
+            case Role.Admin:
+                resource = await createAdmin();
+                resourceType = 'Organization';
+                break;
+            case Role.Receptionist:
+                resource = await createPractitioner();
+                resourceType = 'Practitioner';
+                break;
+        }
+
+        const user = await createUser({
+            password: 'password',
+            resourceType: 'User',
+            email: role + '@beda.software',
+        });
+        const links: any = {};
+        if (role === Role.Patient) {
+            links.patient = {
+                id: resource.id,
+                resourceType: resourceType,
+            };
+        } else if (role === Role.Practitioner || role === Role.Receptionist) {
+            links.practitioner = {
+                id: resource.id,
+                resourceType: resourceType,
+            };
+        } else if (role === Role.Admin) {
+            links.organization = {
+                id: resource.id,
+                resourceType: resourceType,
+            };
+        }
+        const roleData = await ensureSave({
+            name: role,
+            user: {
+                id: user.id!,
+                resourceType: 'User',
+            },
+            links: links,
+            resourceType: 'Role',
+        });
+
+        return {
+            user,
+            role: roleData,
+        };
+    });
+}
+
+describe.each(Object.values(Role))('useSmartApps for role %s', (role) => {
+    test(`Retrieve list of smart apps for ${role} role`, async () => {
+        await initialSetup(role as Role);
+        const result = await renderSmartAppsHook();
+        await waitFor(
+            () => {
+                expect(isSuccess(result.current.appsRemoteData)).toBeTruthy();
+            },
+            { timeout: 30000 },
+        );
+
+        const remoteData = result.current.appsRemoteData.data;
```

Comment:
better not to call the variables like remoteData if it is not remotedata container, here it's just appData. It was a remoteData before accessing .data